### ISSUE TITLE
All Yeelink lights went unavailable few hours ago.

### ISSUE THREAD

**Issue by kodbuzz** [al-one/hass-xiaomi-miot#527]
They were working like a charm with Xiaomi Miot Auto in the past months, just went unavailable hours before without any newly applied upgradations/changes.

Environment:
Raspberry Pi 4B
HA Core:  2022.4.4
Supervisor: supervisor-2022.04.0
HACS: 1.24.5
Xiaomi Miot Auto: v0.5.6

- All lights are operational(on both ble remotes and Mihome app.) with unchanged static local IPs, and a great wifi signal strength.
- Lights are reachable with PING (tested also inside HA container.).
- Lights won't be back available even after several reboots.

Logs:

> Logger: custom_components.xiaomi_miot.light
> Source: custom_components/xiaomi_miot/__init__.py:1242
> Integration: Xiaomi Miot Auto ([documentation](https://github.com/al-one/hass-xiaomi-miot), [issues](https://github.com/al-one/hass-xiaomi-miot/issues))
> First occurred: 15:10:21 (150 occurrences)
> Last logged: 15:24:51

> Dining(yeelink.light.ceiling14): Got MiioException while fetching the state: Unable to discover the device 172.17.1.178, mapping: {'light.on': {'siid': 2, 'piid': 1}, 'light.brightness': {'siid': 2, 'piid': 2}, 'light.color_temperature': {'siid': 2, 'piid': 3}, 'light.mode': {'siid': 2, 'piid': 4}, 'adjust_brightness-3-1': {'siid': 3, 'piid': 1}, 'light_extension.ct_delta': {'siid': 3, 'piid': 2}}, max_properties: 10/6
> 2nd-floor-guest1(yeelink.light.ceiling13): Got MiioException while fetching the state: Unable to discover the device 172.17.1.223, mapping: {'light.on': {'siid': 2, 'piid': 1}, 'light.brightness': {'siid': 2, 'piid': 2}, 'light.color_temperature': {'siid': 2, 'piid': 3}, 'light.mode': {'siid': 2, 'piid': 4}, 'adjust_brightness-3-1': {'siid': 3, 'piid': 1}, 'light_extension.ct_delta': {'siid': 3, 'piid': 2}}, max_properties: 10/6
> 2nd-floor-secondary-bedroom(yeelink.light.ceiling13): Got MiioException while fetching the state: Unable to discover the device 172.17.1.119, mapping: {'light.on': {'siid': 2, 'piid': 1}, 'light.brightness': {'siid': 2, 'piid': 2}, 'light.color_temperature': {'siid': 2, 'piid': 3}, 'light.mode': {'siid': 2, 'piid': 4}, 'adjust_brightness-3-1': {'siid': 3, 'piid': 1}, 'light_extension.ct_delta': {'siid': 3, 'piid': 2}}, max_properties: 10/6
> 2nd-floor-studio(yeelink.light.ceiling18): Got MiioException while fetching the state: Unable to discover the device 172.17.1.198, mapping: {'light.on': {'siid': 2, 'piid': 1}, 'light.brightness': {'siid': 2, 'piid': 2}, 'light.color_temperature': {'siid': 2, 'piid': 3}, 'light.mode': {'siid': 2, 'piid': 4}, 'adjust_brightness-3-1': {'siid': 3, 'piid': 1}, 'light_extension.ct_delta': {'siid': 3, 'piid': 2}}, max_properties: 10/6

**Comment by al-one** [comment]
It is recommended that they be divided into the same subnet.

- https://python-miio.readthedocs.io/en/latest/troubleshooting.html#discover-devices-across-subnets
- https://github.com/rytilahti/python-miio/issues/422#issuecomment-455824152

**Comment by kodbuzz** [comment]
> It is recommended that they be divided into the same subnet.
> 
> * https://python-miio.readthedocs.io/en/latest/troubleshooting.html#discover-devices-across-subnets
> * [Some devices not responding across subnets. rytilahti/python-miio#422 (comment)](https://github.com/rytilahti/python-miio/issues/422#issuecomment-455824152)

Hi, excuse me but I'm not sure if I got it right, my subnet is 172.17.1.0/24, and all devices are in the same subnet, including HA host(RPi-4B), the lights, and all other IoT devices.

Cheers.
Oz

**Comment by al-one** [comment]
It looks ok. I think you may need to use [`miiocli`](https://python-miio.readthedocs.io/en/latest/discovery.html) to test the connection.

```shell
pip install python-miio

miiocli device --ip 172.17.1.178 --token xxxxxx info
```

**Comment by kodbuzz** [comment]
> It looks ok. I think you may need to use [`miiocli`](https://python-miio.readthedocs.io/en/latest/discovery.html) to test the connection.
> 
> ```shell
> pip install python-miio
> 
> miiocli device --ip 172.17.1.178 --token xxxxxx info
> ```


> bash-5.1# miiocli device --ip 172.17.1.178 --token <REDACTED> info
> **Error: Unable to discover the device 172.17.1.178**
> bash-5.1# ping 172.17.1.178
> PING 172.17.1.178 (172.17.1.178): 56 data bytes
> 64 bytes from 172.17.1.178: seq=0 ttl=255 time=34.873 ms
> 64 bytes from 172.17.1.178: seq=1 ttl=255 time=8.813 ms
> 64 bytes from 172.17.1.178: seq=2 ttl=255 time=32.459 ms
> 64 bytes from 172.17.1.178: seq=3 ttl=255 time=56.032 ms
> ^C
> --- 172.17.1.178 ping statistics ---
> 4 packets transmitted, 4 packets received, 0% packet loss
> round-trip min/avg/max = 8.813/33.044/56.032 ms
> bash-5.1# 


That's pretty weird.

**Comment by kodbuzz** [comment]
I'm gonna close this since evidence shows it seems to have nothing to do with Xiaomi Miot Auto, the problem is more likely from upstream or hardware, at least I assume that's the case.

**Comment by kodbuzz** [comment]
Update:

After digging into the issue, across the logbook and history I recalled I replaced a dedicated router for IoT devices about two weeks ago and I've chosen WPA2 only as default security, Yeelink ceiling lights in my house seem to have random chances to stop responding on port 54321 when they get connected with "WPA2 only" since then.

I'm not sure if the problem is router related or from Yeelink, but for better compatibility I changed the wifi security to WPA as a compromise. The problem seems gone for at least 48 hours.
